web_src/js/svg.js

```diff
@@ -207,7 +208,10 @@ export const SvgIcon = {
     if (this.className) {
       classes.push(...this.className.split(/\s+/).filter(Boolean));
     }
-
+    if (this.symbolId) {
+      classes.push('gt-hidden', 'svg-symbol-container');
+      svgInnerHtml = `<symbol id="${this.symbolId}" viewBox="${attrs['^viewBox']}">${svgInnerHtml}</symbol>`;
+    }
```

Comment:
What is this? Is it unused?

Reply:
If it is intentional, could someone explain how this works?

Reply:
It is used, you can grep "symbol-id"

That's what I have proposed to avoid hard-coding the SVG content into Vue while still keeping good performance.

![image](https://github.com/go-gitea/gitea/assets/2114189/a7cf552e-6153-45fa-b38f-13f9cdc8f74d)

Reply:
I see, well it's certainly not an ideal solution, but I guess better then hardcoding  the whole thing.

Reply:
Why it's not ideal? I think it's pretty clear and simple, and performant.

Reply:
Not really.

Before:  4000 SVG nodes.

After:  1 SVG symbol container node + 4000 SVG "use" nodes, and these 4000 nodes don't need to duplicate the SVG XML anymore.


I would say it's pretty ideal.

Reply:
From DOM size standpoint, yes it's better. I think there could even be a general solution with `use` for all vue components, which dynamically appends symbols to one global `<svg>`.